File: lib/test/t_HistogramFactory_std.cxx
//                                               -*- C++ -*-
/**
 *  @file  t_HistogramFactory_std.cxx
 *  @brief The test file of class Histogram for standard methods
 *
 *  Copyright (C) 2005-2013 EDF-EADS-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 *  @author dutka
 *  @date   2010-02-04 16:44:49 +0100 (jeu. 04 févr. 2010)
 */
#include "OT.hxx"
#include "OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int argc, char *argv[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);
  setRandomGenerator();

  try
    {
      UnsignedLong collectionSize(4);
      Histogram::HistogramPairCollection collection(collectionSize);
      collection[0] = HistogramPair(1.0, 0.5);
      collection[1] = HistogramPair(0.7, 1.5);
      collection[2] = HistogramPair(1.2, 3.5);
      collection[3] = HistogramPair(0.9, 2.5);
      Histogram distribution(-1.5, collection);
      UnsignedLong size(10000);
      NumericalSample sample(distribution.getSample(size));
      HistogramFactory factory;
      Distribution estimatedDistribution(factory.build(sample));
      fullprint << "Distribution          =" << distribution << std::endl;
      fullprint << "Estimated distribution=" << estimatedDistribution << std::endl;
      estimatedDistribution = factory.build();
      fullprint << "Default distribution=" << estimatedDistribution << std::endl;
      Histogram estimatedHistogram(factory.buildAsHistogram(sample));
      fullprint << "Histogram          =" << distribution << std::endl;
      fullprint << "Estimated histogram=" << estimatedHistogram << std::endl;
      estimatedHistogram = factory.buildAsHistogram();
      fullprint << "Default histogram=" << estimatedHistogram << std::endl;
    }
  catch (TestFailed & ex)
    {
      std::cerr << ex << std::endl;
      return ExitCode::Error;
    }


  return ExitCode::Success;
}

File: python/test/t_HistogramFactory_std.py
#! /usr/bin/env python

from openturns import *

TESTPREAMBLE()
RandomGenerator().SetSeed(0)

try :
    collectionSize = 4
    collection = HistogramPairCollection(collectionSize)
    collection[0] = HistogramPair(1.0, 0.5)
    collection[1] = HistogramPair(0.7, 1.5)
    collection[2] = HistogramPair(1.2, 3.5)
    collection[3] = HistogramPair(0.9, 2.5)
    distribution = Histogram(-1.5, collection)
    size = 10000
    sample = distribution.getSample(size)
    factory = HistogramFactory()
    estimatedDistribution = factory.build(sample)
    print "distribution=", repr(distribution)
    print "Estimated distribution=", repr(estimatedDistribution)
    estimatedDistribution = factory.build()
    print "Default distribution=", estimatedDistribution
    estimatedHistogram = factory.buildAsHistogram(sample)
    print "Histogram          =", distribution
    print "Estimated histogram=", estimatedHistogram
    estimatedHistogram = factory.buildAsHistogram()
    print "Default histogram=", estimatedHistogram

except :
    import sys
    print "t_HistogramFactory_std.py", sys.exc_type, sys.exc_value
